cocos/renderer/gfx-base/GFXShader.h

```diff
@@ -52,7 +52,7 @@ class CC_DLL Shader : public GFXObject {
 
 protected:
     static uint generateShaderID() noexcept {
-        static uint _idGen = 0;
+        static uint _idGen = 10000;
```

Comment:
Any reason to change the value? And why using `10000`?

Reply:
no particular reason, changed this just so that we don't have 0 a valid id.

Reply:
Got it. Is the validation of  shader id is checked in codes?

Reply:
no, not for now, but definitely can do